### PR TITLE
Improve qv_sat to allow more caller info

### LINKED ISSUE #2040: Provide caller's name in physics_saturation_impl.hpp error message.
Callers of `qv_sat` should pass a label to the function that can be printed when a `check_temperature` check fails. For P3, which calls `qv_sat` in multiple spots, make unique labels for each, e.g. "P3_X", where "X" describes the specific call site.

### PR DESCRIPTION
**Pull request by jgfouca**
Fixes #2040

### CODE REVIEW

**Comment by ambrad** [comment]
These need to be handled, too:
```
./src/physics/shoc/shoc_assumed_pdf_impl.hpp:242:        const Spack esval1_1 = scream::physics::Functions<S,D>::MurphyKoop_svp(Tl1_1,liquid,active_entries);
./src/physics/shoc/shoc_assumed_pdf_impl.hpp:243:        const Spack esval1_2 = scream::physics::Functions<S,D>::MurphyKoop_svp(Tl1_2,liquid,active_entries);
```

File: components/scream/src/physics/share/physics_saturation_impl.hpp [hunk 1]
```diff
@@ -27,22 +27,22 @@ ::check_temperature(const Spack& t_atm, const char* /* func_name */, const Smask
...
 template <typename S, typename D>
 KOKKOS_FUNCTION
 typename Functions<S,D>::Spack
-Functions<S,D>::MurphyKoop_svp(const Spack& t_atm, const bool ice, const Smask& range_mask)
+Functions<S,D>::MurphyKoop_svp(const Spack& t_atm, const bool ice, const Smask& range_mask, const char* caller)
 {
 
   //First check if the temperature is legitimate or not
-  check_temperature(t_atm, "MurphyKoop_svp", range_mask);
+  check_temperature(t_atm, ekat::impl::strlen(caller) == 0 ? "MurphyKoop_svp" : caller, range_mask);
```
Comment (ambrad) [inline components/scream/src/physics/share/physics_saturation_impl.hpp:45]: A default to `nullptr` I think is more natural. Then here you wouldn't call `strlen` but instead just check if `caller == `nullptr` (or, idiomatically, simply `caller ? caller : "MurphyKoop_svp"`).

File: components/scream/src/physics/share/physics_saturation_impl.hpp [hunk 2]
```diff
@@ -84,12 +84,12 @@ Functions<S,D>::MurphyKoop_svp(const Spack& t_atm, const bool ice, const Smask&
...
 KOKKOS_FUNCTION
 typename Functions<S,D>::Spack
-Functions<S,D>::polysvp1(const Spack& t, const bool ice, const Smask& range_mask)
+Functions<S,D>::polysvp1(const Spack& t, const bool ice, const Smask& range_mask, const char* caller)
 {
   // REPLACE GOFF-GRATCH WITH FASTER FORMULATION FROM FLATAU ET AL. 1992, TABLE 4 (RIGHT-HAND COLUMN)
 
   //First check if the temperature is legitimate or not
-  check_temperature(t, "polysvp1", range_mask);
+  check_temperature(t, ekat::impl::strlen(caller) ? "polysvp1" : caller, range_mask);
```
Comment (ambrad) [inline components/scream/src/physics/share/physics_saturation_impl.hpp:92]: Same here re: `caller`. Note that this is a function call, so every call to MurphyKoop_svp will encounter this line, and the args to `check_temperature` will be evaluated always. So we don't want to call `strlen` every time.

**Comment by jgfouca** [comment]
@ambrad , fixed.

**Review by ambrad — APPROVED** [review]

**Comment by ambrad** [comment]
Thanks!

**Comment by AaronDonahue** [comment]
@jgfouca , is there any reason this PR is failing to build on `blake` and `weaver`, but not `mappy`?

**Pull request merged**